### PR TITLE
feat/(Status[Base]Button): add a new Primary variant 

### PR DESCRIPTION
**Pull request by caybro**
- we were missing this variant, needed for https://github.com/status-im/status-desktop/issues/6081
- cleanup *Palette, use a shared impl for the font definitions, they are common to both light/dark palettes

### Checklist

- [x] follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
  - the scope should be the component's name e.g: `feat(StatusListItem): ... `
  - when adding new components, the scope is the module e.g: `feat(StatusQ.Controls): ...`
- [x] add documentation if necessary (new component, new feature)
- [x] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [x] test changes in both light and dark theme?
- [ ] is this a breaking change?
    - [ ] use the dedicated `BREAKING CHANGE` commit message section
    - [ ] resolve breaking changes in [status-desktop](https://github.com/status-im/status-desktop)
        - [ ] (pre-merge) adapt code to breaking changes
        - [ ] (post-merge) update StatusQ submodule pointer
- [ ] test changes in [status-desktop](https://github.com/status-im/status-desktop)

Light version:
Snímek obrazovky z 2022-08-08 14-18-35

Dark version:
Snímek obrazovky z 2022-08-08 14-18-42

### CODE REVIEW

**Review by PascalPrecht — APPROVED** [review]

File: src/StatusQ/Controls/StatusButton.qml [hunk 1]
```diff
@@ -5,13 +5,18 @@ import StatusQ.Core.Theme 0.1
...
-                                                       : Theme.palette.dangerColor3
-    hoverColor:  type === StatusBaseButton.Type.Normal ? Theme.palette.primaryColor2
-                                                       : Theme.palette.dangerColor2
+    normalColor: type === StatusBaseButton.Type.Primary ? Theme.palette.primaryColor1 :
+                                                          type === StatusBaseButton.Type.Normal ? Theme.palette.primaryColor3
+                                                                                                : Theme.palette.dangerColor3
+    hoverColor: type === StatusBaseButton.Type.Primary ? Theme.palette.hoverColor(normalColor) :
+                                                         type === StatusBaseButton.Type.Normal ? Theme.palette.primaryColor2
+                                                                                               : Theme.palette.dangerColor2
     disaledColor: Theme.palette.baseColor2
```
Comment (noeliaSD) [inline src/StatusQ/Controls/StatusButton.qml:14]: Not related to this PR but seen this typo:

```suggestion
    disabledColor: Theme.palette.baseColor2
```

**Review by noeliaSD — APPROVED** [review]
LGTM!

File: src/StatusQ/Controls/StatusButton.qml (hunk 1 above)
Comment (caybro) [reply src/StatusQ/Controls/StatusButton.qml:14]: Existing code and I think @alexandraB99 already fixed it in another PR

File: src/StatusQ/Core/Theme/ThemePalette.qml [hunk 2]
```diff
@@ -6,37 +6,104 @@ QtObject {
...
-    property FontLoader monoFontExtraBold
-    property FontLoader monoFontBlack
-
-    property FontLoader codeFont
-    property FontLoader codeFontThin
-    property FontLoader codeFontExtraLight
-    property FontLoader codeFontLight
-    property FontLoader codeFontMedium
-    property FontLoader codeFontBold
+    property var baseFont: FontLoader {
```
[the remark this answers is not in the document]
Comment (caybro) [reply src/StatusQ/Core/Theme/ThemePalette.qml:9]: Yeah... there's so much stuff that could be better here; like we keep both variants in memory all the time. The new Theme will be C++ based so not a big deal atm :)

**Pull request merged**